software/control/core/core.py

```diff
@@ -1459,7 +1459,8 @@ def run(self):
             self._log.error(f"Operation timed out during acquisition, aborting acquisition!")
             self._log.error(te)
             self.multiPointController.request_abort_aquisition()
-        self.finished.emit()
+        if QApplication.instance() is not None:
+            self.finished.emit()
```

Comment:
@Alpaca233 you mentioned adding a flag instead of checking QApplication.instance(). I think it's useful to do that before merging.